lib/modules/manager/terragrunt/modules.ts

```diff
@@ -68,18 +68,28 @@ export function analyseTerragruntModule(
     dep.currentValue = githubRefMatch.groups.tag;
     dep.datasource = GithubTagsDatasource.id;
   } else if (gitTagsRefMatch?.groups) {
-    dep.depType = 'gitTags';
-    if (gitTagsRefMatch.groups.path.includes('//')) {
+    const { url, host, path, tag } = gitTagsRefMatch.groups;
+    const containsSubDirectory = path.includes('//');
+    if (containsSubDirectory) {
       logger.debug('Terragrunt module contains subdirectory');
-      dep.depName = gitTagsRefMatch.groups.path.split('//')[0];
-      const tempLookupName = gitTagsRefMatch.groups.url.split('//');
-      dep.packageName = tempLookupName[0] + '//' + tempLookupName[1];
+    }
+    dep.depType = 'gitTags';
+    // We don't want to have .git or subdirectory in the depName
+    dep.depName = host + '/' + path.split('//')[0].replace('.git', '');
+    dep.currentValue = tag;
+    dep.datasource = detectGitTagDatasource(url);
+    if (dep.datasource === GitTagsDatasource.id) {
+      if (containsSubDirectory) {
+        const [protocol, hostAndPath] = url.split('//');
+        dep.packageName = protocol + '//' + hostAndPath;
+      } else {
+        dep.packageName = url;
+      }
     } else {
-      dep.depName = gitTagsRefMatch.groups.path.replace('.git', '');
-      dep.packageName = gitTagsRefMatch.groups.url;
+      // The packageName should only contain the path to the repository
+      dep.packageName = path.split('//')[0];
+      dep.registryUrls = [`https://${host}`];
```

Comment:
is it safe to always assume https here 🤔

@rarkins @secustor WDYT?

Reply:
There could be some personal instances hosted on HTTP, but we should be able to use `detectPlatform` to cover these cases. 
Outside of this self-hosted scenario everything should be using HTTPS

Reply:
Just had a look at the `detectPlatroform` utility function. It only returns the platform name, but no information about the protocol used.

Reply:
@secustor is there anything agains adding just both `http` and `https`? 🤔

Reply:
I have thought it should be easy to extend, tough it isn't.

 That would double the API requests.

Let's fix it on https